content/doc/book/pipeline/pipeline-best-practices.adoc

```diff
@@ -0,0 +1,97 @@
+---
+layout: section
+title: Pipeline Best Practices
+---
+ifdef::backend-html5[]
+:notitle:
+:description:
+:author: Alex Taylor
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+:toc:
+endif::[]
+
+= Pipeline Best Practices
+
+== General
+
+=== Pipeline functions as Glue
+
+Make sure to use pipeline steps as a way of wiring together a set of actions rather than the main functionality. 
```

Comment:
What do you mean? In general, I think using Pipeline steps is fine. You can go overboard with highly-specific "publisher" steps, but the real concern IMO is to avoid arbitrary Groovy code. I guess what you mean is using things like `readFile` and then processing that data in the Pipeline rather than doing the processing in a `sh` step and only sending the needed results to the Pipeline? If so, I might reword this to something like this:

> Pipeline is intended to be used to orchestrate your build. Prefer using `sh` steps for arbitrary data processing instead of processing data in your Pipeline script itself using a combination of Pipeline steps and Groovy code.

Also FWIW I think you should capitalize Pipeline throughout the article, since we are talking about Jenkins Pipeline, not the general concept of a pipeline.